app/src/main/java/com/example/makeiteven2/adapters/LevelsAdapter.kt

```diff
@@ -19,20 +17,15 @@ class LevelsAdapter(private var mLevelsList : List<Level> = ArrayList(), private
 
     class ViewHolder(private val view : View,private val context: Context) : RecyclerView.ViewHolder(view){
         private val onTouchAnimation = OnTouchAnimation(context)
-        private lateinit var levelBtn : Button
-        private lateinit var lockStage : View
-        private lateinit var stageNumberLayout : View
-        private lateinit var stageNumberTv : TextView
 
         @SuppressLint("ClickableViewAccessibility")
         fun bind(level : Level, currentStage: Int) {
-            levelBtn = view.btnLevel
-            lockStage= view.ivLock
-            stageNumberLayout = view.stage_number_layout
-            stageNumberTv = view.tvStageNumber
+            val lockStage : View = view.ivLock
```

Comment:
my opinion you just need to use view.ivLock, view.layoutStageNumber, view.tvStageNumber as is dont do variables.
and variable declaration is wrong lockStage is ImageView not View
stage number layout is layoutXXXX not View,
either declare them as members of the class like:
val lockStageImageView = view.ivLock
either just use it like view.ivLock (btw) give it better name